srcs/requirements/frontend/src/navigation.ts

```
@@ -47,13 +54,13 @@ export function navigateTo(view: string, replace: boolean = false): void {
  * @param templateElement The template element whose content will be used to update the host element.
  */
-function updateView(hostElement: HTMLElement, templateElement: HTMLTemplateElement) {
+function renderView(hostElement: HTMLElement, templateElement: HTMLTemplateElement) {
     const clone = document.importNode(templateElement.content, true);
     hostElement.replaceChildren(clone);
 
     const navigationElement = document.getElementById("navigation");
-    if (navigationElement && templateElement.id === "login-template") {
-        navigationElement.innerText = "";
+    if (currentView === "login-template") {
+        navigationElement!.innerText = "";
```

Biome (1.9.4)

[error] Forbidden non-null assertion.



(lint/style/noNonNullAssertion)